usaspending_api/search/v2/views/spending_by_award.py

```diff
@@ -0,0 +1,184 @@
+
+
+GLOBAL_MAP = {
+    "award": {
+        "minimum_db_fields": {"award_id", "piid", "fain", "uri", "type"},
+        "api_to_db_mapping_list": [
+            award_contracts_mapping,
+            award_idv_mapping,
+            loan_award_mapping,
+            non_loan_assistance_award_mapping,
+        ],
+        "award_semaphore": "type",
+        "award_id_fields": ["piid", "fain", "uri"],
+        "internal_id_field": "award_id",
+        "type_code_to_field_map": {
+            **{award_type: award_contracts_mapping for award_type in contract_type_mapping},
+            **{award_type: award_idv_mapping for award_type in idv_type_mapping},
+            **{award_type: loan_award_mapping for award_type in loan_type_mapping},
+            **{award_type: non_loan_assistance_award_mapping for award_type in non_loan_assistance_type_mapping},
+        },
+        "filter_queryset_func": matview_search_filter_determine_award_matview_model,
+    },
+    "subaward": {
+        "minimum_db_fields": {"subaward_number", "piid", "fain", "award_type"},
+        "api_to_db_mapping_list": [contract_subaward_mapping, grant_subaward_mapping],
+        "award_semaphore": "award_type",
+        "award_id_fields": ["award__piid", "award__fain"],
+        "internal_id_field": "subaward_number",
+        "type_code_to_field_map": {"procurement": contract_subaward_mapping, "grant": grant_subaward_mapping},
+        "filter_queryset_func": subaward_filter,
+    },
+}
+
+
+@api_transformations(api_version=settings.API_VERSION, function_list=API_TRANSFORM_FUNCTIONS)
+class SpendingByAwardVisualizationViewSet(APIView):
+    """
+    This route takes award filters and fields, and returns the fields of the filtered awards.
+    """
+
+    endpoint_doc = "usaspending_api/api_contracts/contracts/v2/search/spending_by_award.md"
+
+    @cache_response()
+    def post(self, request):
+        """Return all awards matching the provided filters and limits"""
+        json_request = self.validate_request_data(request.data)
+        self.is_subaward = json_request["subawards"]
+        self.constants = GLOBAL_MAP["subaward"] if self.is_subaward else GLOBAL_MAP["award"]
+        self.filters = json_request["filters"]
+        self.fields = json_request["fields"]
+        self.pagination = {
+            "limit": json_request["limit"],
+            "lower_bound": (json_request["page"] - 1) * json_request["limit"],
+            "page": json_request["page"],
+            "sort_key": json_request.get("sort") or self.fields[0],
+            "sort_order": json_request["order"],
+            "upper_bound": json_request["page"] * json_request["limit"] + 1,
+        }
+
+        if self.if_no_intersection():  # Like an exception, but API response is a HTTP 200 with a JSON payload
+            return Response(self.populate_response(results=[], has_next=False))
+        raise_if_award_types_not_valid_subset(self.filters["award_type_codes"], self.is_subaward)
```

Comment:
```suggestion

        raise_if_award_types_not_valid_subset(self.filters["award_type_codes"], self.is_subaward)
```

Reply:
These sections are logically seperate

Reply:
I disagree

Reply:
While the no_intersection flag can be thought of as "exception" behavior,  the lower group are conditions under which errors are thrown, while the upper is just a case of special responses. We're entirely in subjective territory, but I would call that a difference worthy of a blank line.

Reply:
Both the "no intersection" and the exceptions thrown are special behavior checking for input data which doesn't meet the required conditions (even though the data are technically correct) and short-circuiting the response, preventing the rest of the view logic from executing.

Reply:
It's not required to not be "no intersection", that's a valid, albeit special, case. If I saw an error thrown when trying to run this endpoint and decided to look in the code in this repo, I'm not looking for that line, and diving into that (two line) function would be a waste of my time.

Reply:
There are no "errors" being thrown in this section of code. Exceptions are raised which are caught by the view method and return HTTP4xx responses. You wouldn't see a stack trace even with debug=True.

If the DRF exceptions didn't convert all JSON values to strings the no intersection check would also raise an exception.